File: Python/_014_Longest_Common_Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''

        first = min(strs)
        for i in range(len(first)):
            for s in strs:
                if s[i] != first[i]:
                    return first[:i] if i > 0 else ''
        return first

File: Java/_014_Longest_Common_Prefix/Solution.java
package _014_Longest_Common_Prefix;

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        int min = 0;
	 while (strs.length > 0) {
	    for (String s : strs) 
		if (s.length() == min || s.charAt(min) != strs[0].charAt(min)) 
		    return strs[0].substring(0, min);
	    min++;
	 }
	 return new String("");
    }
}
